roles/ansible_service_broker/tasks/install.yml

```diff
@@ -111,116 +113,110 @@
-                  name: etcd
-                  imagePullPolicy: IfNotPresent
-                  terminationMessagePath: /tmp/termination-log
-                  workingDir: /etcd
-                  args:
-                    - '{{ ansible_service_broker_etcd_image_etcd_path }}'
-                    - --data-dir=/data
-                    - "--listen-client-urls=http://0.0.0.0:2379"
-                    - "--advertise-client-urls=http://0.0.0.0:2379"
-                  ports:
-                    - containerPort: 2379
-                      protocol: TCP
-                  env:
-                    - name: ETCDCTL_API
-                      value: "3"
-                  volumeMounts:
-                    - mountPath: /data
-                      name: etcd
+              - image: "{{ ansible_service_broker_image }}"
+                name: asb
+                imagePullPolicy: IfNotPresent
+                volumeMounts:
+                  - name: config-volume
+                    mountPath: /etc/ansible-service-broker
+                  - name: asb-tls
+                    mountPath: /etc/tls/private
+                  - name: asb-auth-volume
+                    mountPath: /var/run/asb-auth
+                ports:
+                  - containerPort: 1338
+                    protocol: TCP
+                env:
+                - name: BROKER_CONFIG
+                  value: /etc/ansible-service-broker/config.yaml
+                resources: {}
+                terminationMessagePath: /tmp/termination-log
+
+              - image: "{{ ansible_service_broker_etcd_image }}"
+                name: etcd
+                imagePullPolicy: IfNotPresent
+                terminationMessagePath: /tmp/termination-log
+                workingDir: /etcd
+                args:
+                  - "{{ ansible_service_broker_etcd_image_etcd_path }}"
+                  - "--data-dir=/data"
+                  - "--listen-client-urls=http://0.0.0.0:2379"
+                  - "--advertise-client-urls=http://0.0.0.0:2379"
+                ports:
+                - containerPort: 2379
+                  protocol: TCP
+                env:
+                - name: ETCDCTL_API
+                  value: "3"
+                volumeMounts:
+                  - mountPath: /data
+                    name: etcd
               volumes:
                 - name: etcd
                   persistentVolumeClaim:
                     claimName: etcd
+                - name: config-volume
+                  configMap:
+                    name: broker-config
+                    items:
+                    - key: broker-config
+                      path: config.yaml
+                - name: asb-tls
+                  secret:
+                    secretName: asb-tls
+                - name: asb-auth-volume
+                  secret:
+                    secretName: asb-auth-secret
 
-- name: create ansible-service-broker deployment
+
+- name: Create secret for broker auth
```

Comment:
Consider oc_secret module here.

Reply:
I couldn't figure out how to create the secrets this way with the module, would

```yaml
oc_secret:
  name: asb-auth-secret
  namespace: openshift-ansible-service-broker
  state: present
  contents:
    - string_data:
        username: "{{ ansible_service_broker_user }}"
        password: "{{ ansible_service_broker_password }}"
```
be correct?

Reply:
ah, or would it be:
```yaml
oc_secret:
  name: asb-auth-secret
  namespace: openshift-ansible-service-broker
  state: present
  contents:
    - path: /tmp/authsecret
      data:
        username: "{{ ansible_service_broker_user }}"
        password: "{{ ansible_service_broker_password }}"
```

Reply:
Or do I need to base64 encode the contents of data?

Reply:
This last iteration looks right.  No need to base64 as the `oc secret` call will do that for you.

Reply:
Awesome, thanks!